File: 2° BackEnd/app/main/subModels/comment.py
from django.db import models
from main.subModels.user import User
from main.subModels.game import Game
from main.enum.commentEnum import CommentEnum

VOTE_CHOICES = [
    (CommentEnum.PREMIATION.value, CommentEnum.PREMIATION.name),
    (CommentEnum.FUNNY.value, CommentEnum.FUNNY.name),
    (CommentEnum.YES.value, CommentEnum.YES.name),
    (CommentEnum.NO.value, CommentEnum.NO.name)
]

class Comment(models.Model):
    content = models.CharField(max_length = 300, blank=True, null=True)
    vote = models.CharField(max_length = 50, null=True, choices=VOTE_CHOICES)
    publication = models.DateTimeField(auto_now=True)
    user = models.ForeignKey(User, on_delete=models.CASCADE)
    game = models.ForeignKey(Game, on_delete=models.CASCADE)

    def __str__(self):
        return f"\n| Id: {self.id}\n| Content: {self.content}\n| Vote: {self.vote}\n| Publication: {self.publication}\n\n| User: [{self.user} ]\n\n| Game: [{self.game} ]\n\n"

File: 3° FrontEnd/src/app/shared/models/Comment.ts
import { User } from "./User"
import { Game } from "./Game"
import { I_Comment } from "../interfaces/I_Comment"

export class Comment implements I_Comment{
    private _id?: number | undefined
    private _vote?: string | undefined
    private _content?: string
    private _publication?: string
    private _user?: User
    private _game?: Game
    
    constructor(id: number, vote: string, content: string, publication: string, user: any, game: any);
    constructor(content: string, user: User, game: any);
    constructor();
    constructor(...myarray: any[]){
        if (myarray.length === 3) {
            this._content = myarray[0]
            this._user = myarray[1]
            this._game = myarray[2]
            return;
        }
        if (myarray.length === 6) {
            this._id = myarray[0]
            this._vote = myarray[1]
            this._content = myarray[2]
            this._publication = myarray[3]
            this._user = myarray[4]
            this._game = myarray[5]
            return;
        }
    }

    static simpleRefract(comment: Comment) {
        const result = {
            id: comment!.id,
            vote: comment!.vote,
            content: comment!.content,
            publication: comment!.publication,
            user: User.simpleRefract(comment!.user),
            game: Game.simpleRefract(comment!.game)
        }
        return result;
    }

    public get id(): number | undefined {
        return this._id
    }
    public set id(value: number | undefined) {
        this._id = value
    }

    public get vote(): string | undefined {
        return this._vote
    }
    public set vote(value: string | undefined) {
        this._vote = value
    }

    get content(): string | undefined  {
        return this._content
    }
    set content(value: string | undefined )  {
        this._content = value
    }

    get publication(): string | undefined  {
        return this._publication
    }

    set publication(value: string | undefined ) {
        this._publication = value
    }

    public get user(): any | undefined {
        return this._user
    }
    public set user(value: any | undefined) {
        this._user = value
    }

    public get game(): any | undefined {
        return this._game
    }
    public set game(value: any | undefined) {
        this._game = value
    }
}